File: Modules/ServerTypes/Announcement/AnnouncementDivisionBindedForm.ts
﻿import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementDivisionBindedForm {
    AnnouncementWizardId: IntegerEditor;
    DivisionId: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    IsActive: IntegerEditor;
    InsertUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
}

export class AnnouncementDivisionBindedForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementDivisionBinded';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementDivisionBindedForm.init)  {
            AnnouncementDivisionBindedForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(AnnouncementDivisionBindedForm, [
                'AnnouncementWizardId', w0,
                'DivisionId', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'IsActive', w0,
                'InsertUserId', w0,
                'UpdateUserId', w0,
                'DeleteUserId', w0
            ]);
        }
    }
}

File: Modules/HumanResource/Announcement/AnnouncementDivisionBinded/AnnouncementDivisionBindedForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementDivisionBinded")]
[BasedOnRow(typeof(AnnouncementDivisionBindedRow), CheckNames = true)]
public class AnnouncementDivisionBindedForm
{
    public int AnnouncementWizardId { get; set; }
    public int DivisionId { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public short IsActive { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
}